### PR TITLE
manage pagniation when the response is splitted on several pages

### PR DESCRIPTION
**Pull request by yobuntu**

### CODE REVIEW

File: flask_github.py [hunk 1]
```diff
@@ -173,7 +173,17 @@ def request(self, method, resource, **kwargs):
...
         if response.headers['Content-Type'].startswith('application/json'):
-            return response.json()
+            result = response.json()
+            while response.links.get('next'):
+                response = self.session.request(
+                        method, response.links['next']['url'], **kwargs)
+                if not status_code.startswith('2'):
+                    raise GitHubError(response)
+                if response.headers['Content-Type'].startswith('application/json'):
+                    result += response.json()
```
Comment (cenkalti) [inline flask_github.py]: ```type(result) == dict``` How can you add two dicts?

**Comment by yobuntu** [comment]
for what i have see, (in my code, and in the api documentation, only "Requests that return multiple items will be paginated" (see https://developer.github.com/v3/#pagination) and so the result of response.json() should be (and indeed in my test was) a list.

**Comment by cenkalti** [comment]
There are existing users of this package. People may depend on the current behavior. Can we add this functionality without breaking backwards compatibility? (by adding a keyword argument or making it a separate function)

**Comment by yobuntu** [comment]
i added a keyword argument all_pages, default at False for backward compatibility

**Comment by danrschlosser** [comment]
Would you bump the version number?  Pip hasn't pulled down these latest changes

**Comment by cenkalti** [comment]
Done.
https://pypi.python.org/pypi/GitHub-Flask/2.1.0

**Comment by danrschlosser** [comment]
:clap:

**Pull request merged**